crates/rome_js_formatter/src/js/expressions/string_literal_expression.rs

```diff
@@ -10,7 +12,24 @@ impl FormatNode for JsStringLiteralExpression {
         let JsStringLiteralExpressionFields { value_token } = self.as_fields();
 
         let value_token = value_token?;
+        let syntax_node = self.syntax();
+        let parent = syntax_node.parent();
 
-        Ok(format_string_literal_token(value_token, formatter))
+        let needs_parenthesis =
+            if let Some(JsSyntaxKind::JS_EXPRESSION_STATEMENT) = parent.clone().map(|p| p.kind()) {
+                // SAFETY: since parent's `kind` is `JsSyntaxKind::JS_EXPRESSION_STATEMENT`, it would not be `None`.
+                let parent_parent = parent.unwrap().parent();
+                matches!(
+                    parent_parent.map(|p| p.kind()),
+                    Some(JsSyntaxKind::JS_BLOCK_STATEMENT | JsSyntaxKind::JS_MODULE_ITEM_LIST)
+                )
+            } else {
+                false
+            };
```

Comment:
Prettier wraps all string literal expressions in parenthesis, regardless if they are in a block statement or module item list. 

> This is intentional. Prettier wraps any string literal which is in statement position in parentheses because otherwise such strings would get interpreted as directives if they occurred at the top of a function or program, which can change the behavior of the program.

and [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAdKAWABAaiwZgG50SR4BnGVEQkAGhAgAcYBLac5UAQwCdeIAdwAKfBJxTcANoO4BPTgwBGvbmADWcGAGVuAWzgAZVlDjIAZtPJxlqjVu1M1JgObIYvAK42QcPUrgAE0Cgw24oF09uFzgAMQhePW4YNgjkEG5PGAh6EAALGD0pAHU81gonMDhtcXLWADdyuXSwckUQE2teGGFVFySLKx8AK3IAD21XKTgARU8IeEGpawYnXi70pW4AqVymXhMYYtZAmDzkAA4ABlWBa2LVJnT9uC76swYAR3n4XuYJDLkAC0piCQVyvDg31YkN60QGSEsyx81j0rHcXhRU1mPzMiKGDBg22Op3OSAATITVKwpK4AMIQPQI3zkACsuU81gAKtsJEiViB6t4AJJQEKwbRgA4sACCYu0MDk0yW1gAvqqgA)

So this can be simplified to just checking if the parent is a `JsExpressionStatement`. 
```suggestion
            let needs_parenthesis = JsExpressionStatement::can_cast(self.syntax().parent());
```

Reply:
If I don't check `JsExpressionStatement`, these testcase would also change, 
![image](https://user-images.githubusercontent.com/17974631/167128888-6b678959-4e7b-4d53-92a2-f18799c435d0.png)
![image](https://user-images.githubusercontent.com/17974631/167128913-11d70b84-9f7a-492f-8bbe-2c98f0cd0ab8.png)
![image](https://user-images.githubusercontent.com/17974631/167128933-9f6d798c-ef1a-4a08-a031-27e05610b2cd.png)
1. crates/rome_js_formatter/tests/specs/prettier/js/do/do.js.snap
2. crates/rome_js_formatter/tests/specs/prettier/js/decorators/multiple.js.snap
3. crates/rome_js_formatter/tests/specs/prettier/js/record/spread.js.snap

Reply:
Do you think that was alright?

Reply:
although, these test cases are syntax invalid indeed

Reply:
Checking if the parent is a `JsExpressionStatement` is correct but you don't need to check if the grandparent is of a specific kind.